### PR TITLE
🐛 Source Google Ads: Delete required start date

### PR DESCRIPTION
**Pull request by tolik0**
## What

- Delete start_date from required parameters
- Fix pagination

## How
Set the default value for _**start_date**_ as two years ago because incremental queries should include **WHERE** date clause.
Fix pagination to iterate over the pages of response. 

## Recommended reading order
**start_date**:
1. `streams.py`
2. `source.py`

**pagination:**
1. `google_ads.py`

### CODE REVIEW

File: airbyte-integrations/connectors/source-google-ads/acceptance-test-config.yml [hunk 1]
```diff
@@ -22,20 +22,22 @@ acceptance_tests:
...
           - name: "click_view"
             bypass_reason: "Stream not filled yet."
           - name: "unhappytable"
             bypass_reason: "Stream not filled yet."
           - name: "shopping_performance_report"
             bypass_reason: "Stream not filled yet."
+        ignored_fields:
+          accounts:
+            - name: customer.optimization_score_weight
+              bypass_reason: "can be updated, also it is sometimes integer, sometimes float"
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-google-ads/acceptance-test-config.yml:38]: can manually validate typing and update it if needed?

File: airbyte-integrations/connectors/source-google-ads/source_google_ads/streams.py [hunk 2]
```diff
@@ -59,7 +59,7 @@ def chunk_date_range(
     """
     today = pendulum.today(tz=time_zone)
     end_date = min(pendulum.parse(end_date, tz=time_zone), today) if end_date else today
-    start_date = pendulum.parse(start_date, tz=time_zone)
+    start_date = pendulum.parse(start_date, tz=time_zone) if start_date else pendulum.now().subtract(years=2)
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-google-ads/source_google_ads/streams.py:62]: If the start date is not specified, we ought to attempt to retrieve all available data. Alternatively, we can include a comment indicating that the date range query parameters are mandatory, prompting us to assign a default start date if it is absent.

**Review by lazebnyi — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-google-ads/source_google_ads/streams.py (hunk 2 above)
Comment (tolik0) [reply airbyte-integrations/connectors/source-google-ads/source_google_ads/streams.py:62]: Date range in mandatory parameter for the query.

File: airbyte-integrations/connectors/source-google-ads/acceptance-test-config.yml (hunk 1 above)
Comment (tolik0) [reply airbyte-integrations/connectors/source-google-ads/acceptance-test-config.yml]: Updated

File: airbyte-integrations/connectors/source-google-ads/source_google_ads/schemas/accounts.json [hunk 3]
```diff
@@ -42,7 +42,7 @@
       "type": ["null", "number"]
     },
     "customer.optimization_score_weight": {
-      "type": ["null", "number"]
+      "type": ["null", "number", "integer"]
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-google-ads/source_google_ads/schemas/accounts.json:45]: I’m not sure that we support two types for one field

**Review by lazebnyi — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-google-ads/source_google_ads/schemas/accounts.json (hunk 3 above)
Comment (tolik0) [reply airbyte-integrations/connectors/source-google-ads/source_google_ads/schemas/accounts.json:45]: But we also have this type of matching in a few different connectors: 
[source-clockify](https://github.com/airbytehq/airbyte/blob/a277025a8cc6c86ee382353c8542141a44f06405/airbyte-integrations/connectors/source-clockify/source_clockify/schemas/projects.json)
[source-amplitude](https://github.com/airbytehq/airbyte/blob/f5321d18953a4dd5a6f7448ee0bddc23929656cc/airbyte-integrations/connectors/source-amplitude/source_amplitude/schemas/events.json) 
And other
It also doesn't raise errors.

Comment (lazebnyi) [reply airbyte-integrations/connectors/source-google-ads/source_google_ads/schemas/accounts.json:45]: I believe we can update typing during parse response.

Comment (tolik0) [reply airbyte-integrations/connectors/source-google-ads/source_google_ads/schemas/accounts.json]: Fixed

**Review by lazebnyi — APPROVED** [review]

**Comment by oswaldobiolo** [comment]
Hey guys.
After updating my connector version to 0.80, it started showing the following error.
Also, it became very slow after upgrading to this version. Yesterday took more than 4 hours
I rolled back to version 0.7.4 and it works fine again, running in less than 40 minutes

**Comment by tolik0** [comment]
@oswaldobiolo Apologies for the 0.80 issues. A pagination fix is in [#28970](https://github.com/airbytehq/airbyte/pull/28970), expected to merge today.

**Pull request merged**